src/cas/pyodide-cas.ts

```diff
@@ -111,25 +111,87 @@ function usePythonConverter() {
     ['Sqrt', () => 'Sqrt'], // TODO: Replace with power
     ['Root', () => 'Root'], // TODO: Replace with power
     ['EqualEqual', () => 'sympy.Eq'],
+    ['Parentheses', () => ''],
+
+    ['Abs', () => 'sympy.Abs'],
+
+    ['Sin', () => 'sympy.sin'],
+    ['Cos', () => 'sympy.cos'],
+    ['Tan', () => 'sympy.tan'],
+
+    ['Sec', () => 'sympy.sec'],
+    ['Csc', () => 'sympy.csc'],
+    ['Cot', () => 'sympy.cot'],
+
+    ['Arcsin', () => 'sympy.asin'],
+    ['Arccos', () => 'sympy.acos'],
+    ['Arctan', () => 'sympy.atan'],
+
+    ['Asec', () => 'sympy.asec'],
+    ['Acsc', () => 'sympy.acsc'],
+    ['Atan', () => 'sympy.atan'],
+
+    ['Sinh', () => 'sympy.sinh'],
+    ['Cosh', () => 'sympy.cosh'],
+    ['Tanh', () => 'sympy.tanh'],
+
+    ['Sech', () => 'sympy.sech'],
+    ['Csch', () => 'sympy.csch'],
+    ['Coth', () => 'sympy.coth'],
+
+    ['Arsinh', () => 'sympy.asinh'],
+    ['Arcosh', () => 'sympy.acosh'],
+    ['Artanh', () => 'sympy.atanh'],
+
+    ['Asech', () => 'sympy.asech'],
+    ['Acsch', () => 'sympy.acsch'],
+    ['Acoth', () => 'sympy.acoth'],
+
+    ['Pi', () => 'sympy.pi'],
+    ['ImaginaryUnit', () => 'sympy.I'], // TODO: test
+    ['ExponentialE', () => 'sympy.E'], // TODO: test
+    ['GoldenRatio', () => 'sympy.GoldenRatio'], // TODO: test
+    ['EulerGamma', () => 'sympy.EulerGamma'], // TODO: test
+
+    // TODO: MathLive latex serial/parse destroys ln,log, etc? in ExpressionElement.vue line 78
+    // ['log', () => 'sympy.log'],
   ])
 
+  const KnownLatexFunctions = {
+    '\\sin': 'sympy.sin',
+  }
+
   // TODO: Options (rational numbers)
   function expressionToPython(expression: any): string {
     if (Array.isArray(expression)) {
+      // Handle Functions (Add, Power, Sin, etc.)
       expression = expression.slice() // Make a copy so that we can safely modify the expression
       const functionName = expression[0]
-      const toPython = MathJsonToSympy.get(functionName)
-      let pythonFunctionName = toPython ? toPython(expression) : functionName
+      const functionInPython = MathJsonToSympy.get(functionName)
+      let pythonFunctionName = functionInPython ? functionInPython(expression) : functionName
 
       const parameters = []
       for (let i = 1; i < expression.length; i++) {
         parameters.push(expressionToPython(expression[i]))
       }
       return `${pythonFunctionName}(${parameters.join(',')})`
     } else if (typeof expression === 'string') {
-      return encodeName(expression)
+      const constantInPython = MathJsonToSympy.get(expression)
+      // TODO: Handle Units here?
```

Comment:
That's an excellent question. I'll have to give this a thought, since we probably want excellent units support.

Reply:
Okay, here is the deal: I'm not entirely sure what the best way of handling units would be. There are at least three approaches that could be practical

1. Use the [Sympy unit systems](https://docs.sympy.org/latest/modules/physics/units/index.html). I'm not entirely sure if this is very bug-free though.
2. Every unit is a variable. So, `3cm` basically means "3 times 1/100th of a meter". I'm not super sure if Sympy will handle undefined variables in math expressions as nicely as it should.
3. Extract the units and evaluate them separately. So `3 cm / 5 m + 1 m` would be `3/5 + 1`. And the unit of the result would be `cm/m + m`. Though, this might completely break in cases like `3 cm / 5 m + 1 hour`

Reply:
Yea, it is going to need quite a bit of effort. Both in the CAS and the Mathlive side.